platform/FileBase.cpp

```diff
@@ -52,45 +55,53 @@ FileBase::~FileBase()
             p->_next = _next;
         }
     }
-    _mutex->unlock();
+
+    if (_default == this) {
+        _default == NULL;
+    }
 
     if (getPathType() == FilePathType) {
```

Comment:
This part was not locked by _mutex in the original code. Is it locked now by ScopedLock ?
If so, then note that 'remove_filehandle()' uses a lock of its own: filehandle_mutex->lock();
Consider the risk of deadlock.

Reply:
Ta for pointing that out - hadn't considered it. It seems okay, as mbed_retarget.cpp does not call out to `File` objects while holding its filehandle_mutex - it's lower-level.